### PR TITLE
[Search] Updating ES suggesters to account for more phrases (e.g. DOI, author, ..) 

### PR DESCRIPTION
**Pull request by yattias**
- Allow users to search by more phases.
- More specifically, Updating ElasticSearch suggesters to account for more input phrases that would yield results. These new input phrases are DOI, authors, keywords. 
- Assign weights to documents in such a way that documents with higher `hot_score_v2` show up first

### CODE REVIEW

**Review by koutst — APPROVED** [review]
LGTM

File: src/search/documents/paper.py [hunk 1]
```diff
@@ -67,17 +70,68 @@ def should_remove_from_index(self, obj):
...
+        try:
+            oa_data = instance.open_alex_raw_json
+            keywords = [keyword_obj["keyword"] for keyword_obj in oa_data["keywords"]]
+            joined_kewords = " ".join(keywords)
+
+            phrases.append(joined_kewords)
+            phrases.extend(keywords)
+
+        except:
+            pass
```
Comment (gzurowski) [inline src/search/documents/paper.py:102]: Nite: Avoid bare exceptions:

```suggestion
        except Exceptions:
            pass
```

See: https://stackoverflow.com/questions/54948548/what-is-wrong-with-using-a-bare-except

File: src/search/documents/paper.py [hunk 2]
```diff
@@ -67,17 +70,68 @@ def should_remove_from_index(self, obj):
...
+                f"{author['first_name']} {author['last_name']}"
+                for author in authors_list
+                if author["first_name"] and author["last_name"]
+            ]
+            all_authors_as_str = ", ".join(author_names_only)
+
+            phrases.append(all_authors_as_str)
+            phrases.extend(author_names_only)
+        except:
+            pass
```
Comment (gzurowski) [inline src/search/documents/paper.py:117]: Nit: Avoid bare exception:

```suggestion
        except Exception:
            pass
```

File: src/search/documents/post.py [hunk 3]
```diff
@@ -57,16 +59,53 @@ class Django:
...
+            all_authors_as_str = ", ".join(author_names_only)
+            created_by = (
+                instance.created_by.first_name + " " + instance.created_by.last_name
+            )
+
+            phrases.append(all_authors_as_str)
+            phrases.append(created_by)
+            phrases.extend(author_names_only)
+        except:
+            pass
```
Comment (gzurowski) [inline src/search/documents/post.py:91]: Bare exception

```suggestion
        except Exception:
            pass
```

**Review by gzurowski — APPROVED** [review]
LGTM 👍

File: src/search/documents/paper.py (hunk 1 above)
Comment (yattias) [reply src/search/documents/paper.py:102]: Interesting. Didn't know about bare excepts!

**Pull request merged**